pika/adapters/select_connection.py

```diff
@@ -27,20 +27,25 @@
 ERROR = 0x0008
 
 
-if pika.compat.PY2:
-    _SELECT_ERROR = select.error
-else:
-    # select.error was deprecated and replaced by OSError in python 3.3
-    _SELECT_ERROR = OSError
 
+_SELECT_ERROR_CHECKERS = {
+    select.error : lambda e: e.args[0] == errno.EINTR,
+    OSError      : lambda e: e.errno == errno.EINTR,
+    IOError      : lambda e: e.errno == errno.EINTR,
+    }
 
-def _get_select_errno(error):
-    if pika.compat.PY2:
-        assert isinstance(error, select.error), repr(error)
-        return error.args[0]
-    else:
-        assert isinstance(error, OSError), repr(error)
-        return error.errno
+if pika.compat.PY3:
+    # InterruptedError is undefined in PY2
+    #pylint: disable=E0602
+    _SELECT_ERROR_CHECKERS[InterruptedError] = lambda e: True
```

Comment:
I guess we do.
In python3.4 on Windows;
>>> e = InterruptedError()
>>> type(e)
<class 'InterruptedError'>
>>> isinstance(e, OSError)
True

Also please search for nose.proxy.KeyError: <class 'InterruptedError'> in the failed logs at
https://travis-ci.org/pika/pika/builds/79073409